### PR TITLE
Fix reference implementation for nested local functions

### PR DESCRIPTION
**Pull request by xadupre**
### Description
Propagation model local functions to subgraphs.

### Motivation and Context
ReferenceEvaluator fails at calling a local function from a subgraph in a local functions.

### CODE REVIEW

File: onnx/test/reference_evaluator_test.py [hunk 1]
```diff
@@ -5339,6 +5340,163 @@ def test_regex_invalid_pattern(self):
...
+            opset_imports=[
+                make_operatorsetid("", 15),
+                make_operatorsetid("this", 1),
+            ],
+            functions=[func_def_add, func_def],
+        )
+
+        feeds = {"X": np.array([-5], dtype=np.float32)}
+        oinf = ReferenceEvaluator(model_def)
+        expected = oinf.run(None, feeds)
```
Comment (justinchuby) [inline onnx/test/reference_evaluator_test.py:5491]: Is this expected or actual?

**Review by justinchuby — APPROVED** [review]

File: onnx/test/reference_evaluator_test.py (hunk 1 above)
Comment (xadupre) [reply onnx/test/reference_evaluator_test.py:5491]: I was expecting the inliner to work. So that would have been the expected value for the inlined model.

Comment (gramalingam) [reply onnx/test/reference_evaluator_test.py:5491]: This PR https://github.com/onnx/onnx/pull/5841 hopefully resolves the issue. The problem was that subgraphs were not being recursively processed.

File: onnx/test/reference_evaluator_test.py [hunk 2]
```diff
@@ -5339,6 +5340,163 @@ def test_regex_invalid_pattern(self):
         with self.assertRaises(ValueError):
             ref.run(None, {"X": np.array(["x"])})
 
+    def test_a_function_calling_a_function_once(self):
+        X = make_tensor_value_info("X", TensorProto.FLOAT, ["N"])
+        out = make_tensor_value_info("output", TensorProto.FLOAT, ["N"])
+        Z = make_tensor_value_info("output", TensorProto.FLOAT, ["N"])
```
Comment (gramalingam) [inline onnx/test/reference_evaluator_test.py:5346]: nit: using same name (eg., `output = make_tensor_value_info("output", TensorProto.FLOAT, ["N"])`) would be helpful in the above two lines.

**Review by gramalingam — APPROVED** [review]

File: onnx/test/reference_evaluator_test.py (hunk 2 above)
Comment (xadupre) [reply onnx/test/reference_evaluator_test.py:5346]: done

**Pull request merged**